contracts/profit/src/state/idle.rs

```diff
@@ -43,66 +46,75 @@ impl Idle {
         env: &Env,
         querier: &QuerierWrapper<'_>,
         account: B,
+        native: Vec<CoinDTO<Native>>,
     ) -> ContractResult<PlatformResponse>
     where
         B: BankAccount,
     {
-        let state_batch = self.enter(env.block.time, querier)?;
+        let state_response: PlatformResponse =
+            PlatformResponse::messages_only(self.enter(env.block.time, querier)?);
 
-        let balance_nls: Coin<Nls> = account.balance()?;
+        // TODO Generalize to handle all native currencies
```

Comment:
to alleviate this:
1. revert changes in 'finance::Group'
2. pass 'native' as CoinDTO<Nls>
3. refactor CoinToDTO to split the coins into 'Nls' and 'non-Nls'